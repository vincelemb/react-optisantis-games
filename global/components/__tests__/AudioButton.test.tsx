import React from 'react';
import { shallow } from 'enzyme';
import { AudioButton } from '..';

describe('<AudioButton />', () => {
    const mockCallBack = jest.fn();
    const props = { id: 'music', audioFile: 'audio-file-test.mp3', isPlaying: true, toggleMusic: mockCallBack };
    const wrapper = shallow(<AudioButton {...props} />);
    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('Test click event', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper).toMatchSnapshot();
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
    it('Test audiofile extension', () => {
        let parameter = props.audioFile;
        let extension = '.mp3';
        expect(parameter.endsWith(extension)).toBe(true);
    });
    it('Test id of button ', () => {
        let parameter = props.id;
        let extension = ['music', 'audio'];
        expect(parameter === extension[0] || parameter === extension[1]).toBe(true);
    });
});
