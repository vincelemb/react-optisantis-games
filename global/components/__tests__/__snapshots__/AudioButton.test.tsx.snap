// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioButton /> Test click event 1`] = `
<button
  className="_bg-white _rounded-rounded _w-xxl _h-xxl _border-none _p-xs _cursor-pointer _outline-none"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <SoundOnSvg
    svgWidth="40"
  />
  <Audio
    audioFile="audio-file-test.mp3"
    id="music"
  />
</button>
`;

exports[`<AudioButton /> should render 1`] = `
<button
  className="_bg-white _rounded-rounded _w-xxl _h-xxl _border-none _p-xs _cursor-pointer _outline-none"
  onClick={[MockFunction]}
>
  <SoundOnSvg
    svgWidth="40"
  />
  <Audio
    audioFile="audio-file-test.mp3"
    id="music"
  />
</button>
`;
